numismatic/feeds/base.py

```diff
@@ -71,8 +73,18 @@ class WebsocketClient(abc.ABC):
     raw_stream = attr.ib(default=None)
     raw_interval = attr.ib(default=1)
 
+    @classmethod
+    async def _connect(cls, wss_url=None):
+        if wss_url is None:
+            wss_url = cls.wss_url
+        logger.info(f'Connecting to {wss_url!r} ...')
+        ws = await websockets.connect(wss_url)
+        if hasattr(cls, 'on_connect'):
+            await cls.on_connect(ws)
+        return ws
+
     @abc.abstractmethod
-    async def listen(self, symbol):
+    async def _subscribe(self, symbol, channel=None, wss_url=None):
```

Comment:
You need to go over the collector strategy with me. I would definitely like to start seeing us use them.